### DISCUSSION TITLE
Resolve media URL error : An unexpected error occurred during media resolution.

### DISCUSSION THREAD

**Discussion by Orics** [mozilla/hubs#5874] [Hubs Cloud Questions]
I'm trying to add the quiz function. But I had a problem when put URL into hub room.
- Problem description: 
 -- I had a problem when put media URL from chat box into room. 
 -- I tried many media URL, something worked, something didn't work.

- Step by step:
 -- Enter VR room by URL: https://larchiveum.link/gXSnL7K/?displayName=KoKo&avatarId=Q8LYlzk
 -- Open chat box
 -- Enter this media URL link: http://larchiveum-content.360fairs.com/quiz?quizId=4&token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NDksImVtYWlsIjoiamJzaGluQGdlbWlzby5jb20iLCJtZXRob2QiOjUsInR5cGUiOiI1IiwiaWF0IjoxNjcyNzI4OTYzLCJleHAiOjE2NzQ4ODg5NjN9.H_C9gqIXF8vsefyCbLruy8ca8MfLjP4cUnIz-ol81gg
 -- Drag and drop media URL into room.
 -- > Broken media link ...

- Screen captures: 
![image](https://user-images.githubusercontent.com/32889534/210317048-6972730a-3b5d-44f4-bf31-a558c194312e.png)
![image](https://user-images.githubusercontent.com/32889534/210318430-852c9e7b-093c-4017-986e-efc3ed510f71.png)
![image](https://user-images.githubusercontent.com/32889534/210318542-96774bcd-3e17-4e30-b113-06d275b4f552.png)
![image](https://user-images.githubusercontent.com/32889534/210318660-6ed895a5-0b09-432e-a959-8de3340ef4a6.png)
![image](https://user-images.githubusercontent.com/32889534/210318775-c62e8f9d-9bfb-4104-9737-f0177b754c56.png)

**Comment by rawnsley** [comment]
For background: external URLs are resolved by the backend server (reticulum) rather than by the client's web browser. This is primarily required to work around CORS restrictions, but also allows for things like thumbnail generation for the placeholder image.

Therefore the failure is actually happening on the server side and then a generic message is passed back to the web browser. Hopefully there is a more detailed message written to the log of the backend server that would help you diagnose the problem. Here's how you connect to the backend servers: https://hubs.mozilla.com/docs/hubs-cloud-accessing-servers.html. Something to look out for is that you may have two servers, one for VoIP and one for everything else, including reticulum, which provides the media link resolution service I think.

**Reply by Orics** [reply]
Thank you. I will try to read log of Reticulum

**Reply by Orics** [reply]
@rawnsley 
This is log of backend server. [log.txt](https://github.com/mozilla/hubs/files/10335623/log.txt)
This error also occurs with hub demo room (https://hubs.mozilla.com/AxV22wu/hubs-demo-room).

**Reply by rawnsley** [reply]
Nothing obvious in that log unfortunately. If you watch the log and then load the failing room you might see something. It will look at bit like this line:

`Jan 03 09:48:09 lucid-giant.larchiveum.link bash[1935]: reticulum.default@hubs-dev-01(O): 09:48:09.381 request_id=FzbDhWlyy679WWEAAjxS [info] POST /api/v1/media
`
